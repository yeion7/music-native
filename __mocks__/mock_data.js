export const status = {
  showPlayer: false,
  expand: false,
  isLoading: false,
  isPlaying: false,
  position: null,
  duration: null
};

export const song = {
  id: "0dMDz3n9qLfRXBNlRhK5U7",
  name: "Joe Tex, These Taming Blues",
  duration_ms: 330813,
  preview_url: "https://p.scdn.co/mp3-preview/45d2ad042bf08d7951076097a1f7f15677ab7a89?cid=null",
  artists: [
    {
      external_urls: {
        spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
      },
      href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
      id: "57kIMCLPgkzQlXjblX7XXP",
      name: "Phosphorescent",
      type: "artist",
      uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
    }
  ],
  album: {
    album_type: "album",
    artists: [
      {
        external_urls: {
          spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
        },
        href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
        id: "57kIMCLPgkzQlXjblX7XXP",
        name: "Phosphorescent",
        type: "artist",
        uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
      }
    ],
    external_urls: {
      spotify: "https://open.spotify.com/album/0TticAuqNj26EhgT2Em3qD"
    },
    href: "https://api.spotify.com/v1/albums/0TticAuqNj26EhgT2Em3qD",
    id: "0TticAuqNj26EhgT2Em3qD",
    images: [
      {
        height: 640,
        url: "https://i.scdn.co/image/ccd62a1aa5c66d581622a1be2c86cfd0299a3b45",
        width: 640
      },
      {
        height: 300,
        url: "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
        width: 300
      },
      {
        height: 64,
        url: "https://i.scdn.co/image/1ea5645d08a452f50c9720aad99d4ea6e8cdd8b6",
        width: 64
      }
    ],
    name: "Aw Come Aw Wry",
    type: "album",
    uri: "spotify:album:0TticAuqNj26EhgT2Em3qD"
  },
  track_number: 3
};

export const playlist = [
  {
    id: "0dMDz3n9qLfRXBNlRhK5U7",
    name: "Joe Tex, These Taming Blues",
    duration_ms: 330813,
    preview_url: "https://p.scdn.co/mp3-preview/45d2ad042bf08d7951076097a1f7f15677ab7a89?cid=null",
    artists: [
      {
        external_urls: {
          spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
        },
        href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
        id: "57kIMCLPgkzQlXjblX7XXP",
        name: "Phosphorescent",
        type: "artist",
        uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
      }
    ],
    album: {
      album_type: "album",
      artists: [
        {
          external_urls: {
            spotify: "https://open.spotify.com/artist/57kIMCLPgkzQlXjblX7XXP"
          },
          href: "https://api.spotify.com/v1/artists/57kIMCLPgkzQlXjblX7XXP",
          id: "57kIMCLPgkzQlXjblX7XXP",
          name: "Phosphorescent",
          type: "artist",
          uri: "spotify:artist:57kIMCLPgkzQlXjblX7XXP"
        }
      ],
      external_urls: {
        spotify: "https://open.spotify.com/album/0TticAuqNj26EhgT2Em3qD"
      },
      href: "https://api.spotify.com/v1/albums/0TticAuqNj26EhgT2Em3qD",
      id: "0TticAuqNj26EhgT2Em3qD",
      images: [
        {
          height: 640,
          url: "https://i.scdn.co/image/ccd62a1aa5c66d581622a1be2c86cfd0299a3b45",
          width: 640
        },
        {
          height: 300,
          url: "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
          width: 300
        },
        {
          height: 64,
          url: "https://i.scdn.co/image/1ea5645d08a452f50c9720aad99d4ea6e8cdd8b6",
          width: 64
        }
      ],
      name: "Aw Come Aw Wry",
      type: "album",
      uri: "spotify:album:0TticAuqNj26EhgT2Em3qD"
    },
    track_number: 3
  }
];

export const albums = [
  {
    id: "1bypUQ8qclz4N9ocZgAmPj",
    name: "Aw Man Oh Dear",
    image: "https://i.scdn.co/image/172a34d390848ba1615374fa4f77c304de01a919",
    artists: {
      external_urls: {
        spotify: "https://open.spotify.com/artist/77PtLt3o5UjeAT38mVlGaQ"
      },
      href: "https://api.spotify.com/v1/artists/77PtLt3o5UjeAT38mVlGaQ",
      id: "77PtLt3o5UjeAT38mVlGaQ",
      name: "Aw Man Oh Dear",
      type: "artist",
      uri: "spotify:artist:77PtLt3o5UjeAT38mVlGaQ"
    },
    url_songs: "https://api.spotify.com/v1/albums/1bypUQ8qclz4N9ocZgAmPj",
    type: "album"
  }
];
