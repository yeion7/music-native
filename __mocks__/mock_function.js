export const Mock = jest.fn();
