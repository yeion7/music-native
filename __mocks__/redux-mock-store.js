import configureMockStore from "redux-mock-store";

const middlewares = [];
const mockStore = configureMockStore(middlewares);

export default mockStore;
