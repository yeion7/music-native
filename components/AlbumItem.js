/* @flow weak */

import React from "react";

import {
  Icon,
  List,
  ListItem,
  Thumbnail,
  Body,
  Right,
  Text
} from "native-base";

import { formattedTime } from "../lib/time";

const AlbumItem = ({
  id,
  name,
  image,
  artists,
  url_songs,
  type,
  handlePress
}) => (
  <ListItem key={id} onPress={() => handlePress(url_songs)}>
    <Thumbnail square size={80} source={{ uri: image }} />
    <Body>
      <Text>{name}</Text>
      <Text note>{artists.name}</Text>
      <Text note>{type}</Text>
    </Body>
    <Right>
      <Icon name="md-play" />
    </Right>
  </ListItem>
);

export default AlbumItem;
