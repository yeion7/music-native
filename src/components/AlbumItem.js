/* @flow weak */

import React from "react";

import {
  Icon,
  List,
  ListItem,
  Thumbnail,
  Body,
  Right,
  Text
} from "native-base";

import { formattedTime } from "../utils/time";

const AlbumItem = ({
  id,
  name,
  image,
  artists,
  url_songs,
  type,
  handlePressAlbum
}) => (
  <ListItem onPress={handlePressAlbum}>
    <Thumbnail square size={80} source={{ uri: image }} />
    <Body>
      <Text>{name}</Text>
      <Text note>{artists.name}</Text>
      <Text note>{type}</Text>
    </Body>
    <Right>
      <Icon name="md-play" />
    </Right>
  </ListItem>
);

export default AlbumItem;
