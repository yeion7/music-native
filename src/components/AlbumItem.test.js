import React from "react";
import renderer from "react-test-renderer";
import "react-native";

import { Mock } from "../../__mocks__/mock_function";
import { albums } from "../../__mocks__/mock_data";

import AlbumItem from "./AlbumItem";

describe("renders test ", () => {
  it("render AlbumItem component", () => {
    const tree = renderer
      .create(<AlbumItem {...albums[0]} handlePress={Mock} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
