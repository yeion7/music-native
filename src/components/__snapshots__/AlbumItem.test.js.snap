// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test  render AlbumItem component 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderBottomWidth": 0.5,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 15,
        "paddingBottom": 13,
        "paddingRight": 15,
        "paddingTop": 13,
      }
    }
    virtual={undefined}
  >
    <Image
      size={80}
      source={
        Object {
          "uri": "https://i.scdn.co/image/cc9f4c667dc6fa65839c121262a7e7ba5dd49547",
        }
      }
      square={true}
      style={
        Object {
          "borderRadius": 0,
          "height": 56,
          "width": 56,
        }
      }
      virtual={undefined}
    />
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "flex": 1,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        Aw Come Aw Wry
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        Phosphorescent
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        note={true}
        style={
          Object {
            "color": "#808080",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "200",
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
        virtual={undefined}
      >
        album
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 0.28,
          "paddingBottom": null,
          "paddingLeft": null,
          "paddingRight": null,
          "paddingTop": null,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
</View>
`;
